### PR TITLE
✨ Implement pullToRefresh feature

### PR DESCRIPTION
**Pull request by dev-drprasad**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

**Screenshots**
screen shot 2018-06-12 at 1 26 45 pm
screen shot 2018-06-14 at 12 29 51 pm

Closes #3263

### CODE REVIEW

File: src/components/infiniteScroll/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // @flow
 import * as React from 'react';
+import PullToRefresh from 'pulltorefreshjs';
```
Comment (dev-drprasad) [inline src/components/infiniteScroll/index.js:3]: @mxstbr any particular reason, why we need to implement only in `InfiniteScroll` component ??

File: src/views/dashboard/components/threadFeed.js [hunk 2]
```diff
@@ -304,6 +304,7 @@ class ThreadFeed extends React.Component<Props, State> {
           useWindow={false}
           initialLoad={false}
           scrollElement={scrollElement}
+          showPTRBefore={'#scroller-for-inbox'}
```
Comment (dev-drprasad) [inline src/views/dashboard/components/threadFeed.js]: @mxstbr we receive selector before where we should show pullToRequest element.  what do you say ??

Comment (dev-drprasad) [reply src/views/dashboard/components/threadFeed.js]: @mxstbr what are other places we need pullToRequest other than `threadFeed` ??

Comment (mxstbr) [reply src/views/dashboard/components/threadFeed.js]: No other places we need to have pull to refresh, for now just the thread feed is good enough.

File: src/components/infiniteScroll/index.js [hunk 3]
```diff
@@ -28,23 +30,32 @@ export default class InfiniteScroll extends React.Component<Props> {
...
     this.scrollListener = this.scrollListener.bind(this);
+    this.pullToRefresh = null;
   }
 
   componentDidMount() {
     this.pageLoaded = this.props.pageStart;
     this.attachScrollListener();
+
+    if (this.props.showPTRBefore) {
+      this.pullToRefresh = PullToRefresh.init({
```
Comment (dev-drprasad) [inline src/components/infiniteScroll/index.js:57]: @mxstbr what is expected behaviour for pullToRefresh ?? currently it is default to `window.location.reload()`. should we use `setState` to update thread instead ??

Comment (mxstbr) [reply src/components/infiniteScroll/index.js:57]: It should call `this.props.refetch` if it exists, which should be passed in from the threadFeed component via Apollo's `this.props.data.refetch`!

Comment (dev-drprasad) [reply src/components/infiniteScroll/index.js]: @mxstbr what is `mobile` directory in codebase ?? its looks similar to `src`

**Comment by dev-drprasad** [comment]
@mxstbr what is `mobile` directory in codebase ?? its looks similar to `src`

**Comment by mxstbr** [comment]
It's the mobile ReactNative apps, but those automatically have pull to refresh. :blush:

File: src/components/infiniteScroll/index.js [hunk 4]
```diff
@@ -22,29 +25,41 @@ export default class InfiniteScroll extends React.Component<Props> {
...
   componentDidMount() {
     this.pageLoaded = this.props.pageStart;
+    const { showPTRBefore, refetch } = this.props;
     this.attachScrollListener();
+
+    if (showPTRBefore) {
+      this.pullToRefresh = PullToRefresh.init({
+        mainElement: showPTRBefore,
+        triggerElement: showPTRBefore,
+        onRefresh: refetch ? refetch : () => window.location.reload(),
```
Comment (dev-drprasad) [inline src/components/infiniteScroll/index.js:60]: @mxstbr why i am getting parse Error with this syntax `onRefresh: refetch || () => window.location.reload(),` ??

**Comment by dev-drprasad** [comment]
@mxstbr ready for review 😎. I tested in desktop chrome with mobile view. i haven't tested in real mobile because lot of issues while accessing app via LAN 😓

**Comment by brianlovin** [comment]
I know we're super late on this, but I wanted to see if we could get this in. I pulled and rebased on alpha, but there seems to be a couple lingering bugs in this branch:
- If you scroll down in the feed a little ways, then scroll back up, it triggers the pull-to-refresh even though you're not at the top of the feed
- Scrolling the feed in general feels quite stuttery (not sure if it's just my iOS simulator or if it's the app itself?)

**Comment by dev-drprasad** [comment]
@brianlovin I fixed refreshing happening at middle (i guess). can you rebase again and check ? ( i dont have any simulator)
I am using chrome desktop browser. I am not seeing any stutter. is it possible to install ios simulator you are using  in my laptop ?? I am using ubuntu

**Pull request closed without merging**